### PR TITLE
Base skeletons for policy cmdlets and coversion helpers

### PR DESCRIPTION
**Pull request by sudhakar4171**
This contains base skeletons for policy cmdlets and coversion helpers. Tracing and exception handling will be done in next checkin.

### CODE REVIEW

File: src/ResourceManager/RecoveryServices.Backup/CmdletParameterHelpMessages.cs [hunk 1]
```diff
@@ -33,6 +33,16 @@ internal static class Container
...
+            public const string WorkloadType = "Workload type of the resource (for example: AzureVM, WindowsServer).";
+            public const string BackupManagementType = "Backup Management type of the resource (for example: MAB, DPM).";
+        }
+
+        internal static class Policy
+        {
+            public const string Name = "Name of the Policy that is being managed.";
+            public const string RetentionPolicy = "Retention Policy object for the policy.";
+            public const string SchedulePolicy = "Schedule Policy object for the policy.";
+            public const string ProtectionPolicy = "Protection policy object.";
```
Comment (sudhakar4171) [inline src/ResourceManager/RecoveryServices.Backup/CmdletParameterHelpMessages.cs]: we will get all these messages reviewed by PM at the end

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Conversions/ConversionHelpers.cs [hunk 2]
```diff
@@ -51,5 +52,66 @@ public static List<AzureRmRecoveryServicesContainerBase> GetContainerModelList(I
...
+        #endregion
+
+        #region policy
+        public static AzureRmRecoveryServicesPolicyBase GetPolicyModel(ProtectionPolicyResource hydraResponse)
+        {
+            AzureRmRecoveryServicesPolicyBase policyModel = null;
+
+            if(hydraResponse == null || hydraResponse.Properties == null)
+            {
+                throw new ArgumentException("Empty policy Hydra response");
```
Comment (AditTulasi) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Conversions/ConversionHelpers.cs]: Let's have all exception and warning messages in one place. I have added a file for the same. Please make sure you move it to that file after my PR gets merged.

File: src/ResourceManager/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/GetAzureRmRecoveryServicesPolicy.cs [hunk 3]
```diff
@@ -26,23 +26,52 @@ namespace Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets
...
         public string Name { get; set; }
 
+        [Parameter(ParameterSetName = WorkloadParamSet, Mandatory = true, HelpMessage = ParamHelpMsg.Common.WorkloadType)]
+        [Parameter(ParameterSetName = WorkloadBackupMangementTypeParamSet, Mandatory = true, HelpMessage = ParamHelpMsg.Common.WorkloadType)]
+        [ValidateNotNullOrEmpty]
+        public WorkloadType WorkloadType { get; set; }
+
+        [Parameter(ParameterSetName = WorkloadBackupMangementTypeParamSet, Mandatory = false, HelpMessage = ParamHelpMsg.Common.BackupManagementType)]
+        [ValidateNotNullOrEmpty]
+        public BackupManagementType BackupManagementType { get; set; }
```
Comment (dragonfly91) [inline src/ResourceManager/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/GetAzureRmRecoveryServicesPolicy.cs]: Please test these param set behaviors to ensure they work as per the PM spec.

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Conversions/ConversionHelpers.cs (hunk 2 above)
Comment (sudhakar4171) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Conversions/ConversionHelpers.cs]: sure.

File: src/ResourceManager/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/NewAzureRmRecoveryServicesPolicy.cs [hunk 4]
```diff
@@ -18,6 +18,8 @@
 using System.Threading.Tasks;
 using System.Collections.Generic;
 using System.Management.Automation;
+using Microsoft.Azure.Commands.RecoveryServices.Backup.Helpers;
+using Microsoft.Azure.Management.RecoveryServices.Backup.Models;
```
Comment (pragrawa) [inline src/ResourceManager/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/NewAzureRmRecoveryServicesPolicy.cs]: Sort the usings

File: src/ResourceManager/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/NewAzureRmRecoveryServicesPolicy.cs [hunk 5]
```diff
@@ -55,10 +57,21 @@ public override void ExecuteCmdlet()
...
-                {ContainerParams.Name, Name},             
+                {PolicyParams.PolicyName, Name},
+                {PolicyParams.WorkloadType, WorkloadType},
+                {PolicyParams.BackupManagementType, BackupManagementType},
+                {PolicyParams.RetentionPolicy, RetentionPolicy},
+                {PolicyParams.SchedulePolicy, SchedulePolicy},                
             }, HydraAdapter);
 
-            IPsBackupProvider psBackupProvider = providerManager.GetProviderInstance(ContainerType.AzureVM);
+            IPsBackupProvider psBackupProvider = providerManager.GetProviderInstance(WorkloadType, BackupManagementType);
```
Comment (pragrawa) [inline src/ResourceManager/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/NewAzureRmRecoveryServicesPolicy.cs]: Check the behavior when PSuser doesn't pass BackupManagementType

File: src/ResourceManager/RecoveryServices.Backup/CmdletParameterHelpMessages.cs [hunk 6]
```diff
@@ -33,6 +33,16 @@ internal static class Container
         internal static class Common
         {
             public const string Vault = "The Azure Backup vault object which is the parent resource.";
+            public const string WorkloadType = "Workload type of the resource (for example: AzureVM, WindowsServer).";
```
Comment (pragrawa) [inline src/ResourceManager/RecoveryServices.Backup/CmdletParameterHelpMessages.cs]: Remove WindowsServer from WorkloadType. WindowsServer should fall under ContainerType

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Conversions/RetentionPolicyConversions.cs [hunk 7]
```diff
@@ -25,11 +25,30 @@ namespace Microsoft.Azure.Commands.RecoveryServices.Backup.Helpers
...
+
+            // TBD
+
+            return ltrPolicy;            
+        }
+
+        public static AzureRmRecoveryServicesLongTermRetentionPolicy GetPSSimpleRetentionPolicy(
+           SimpleRetentionPolicy hydraRetPolicy)
+        {
             throw new NotImplementedException();
```
Comment (pragrawa) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Conversions/RetentionPolicyConversions.cs]: Define new common exception saying NotSupportedAction

**Pull request merged**